crates/matrix-sdk/src/send_queue/upload.rs

```diff
@@ -348,36 +350,58 @@ impl QueueStorage {
     }
 
     /// Consumes a finished upload of a thumbnail and queues the file upload.
-    pub(super) async fn handle_dependent_file_upload_with_thumbnail(
+    #[allow(clippy::too_many_arguments)]
+    pub(super) async fn handle_dependent_file_or_thumbnail_upload(
         &self,
         client: &Client,
         next_upload_txn: OwnedTransactionId,
         parent_key: SentRequestKey,
         content_type: String,
         cache_key: MediaRequestParameters,
         event_txn: OwnedTransactionId,
+        depends_on_thumbnail: bool,
     ) -> Result<(), RoomSendQueueError> {
-        // The thumbnail has been sent, now transform the dependent file upload request
-        // into a ready one.
+        // The previous file or thumbnail has been sent, now transform the dependent
+        // file or thumbnail upload request into a ready one.
         let sent_media = parent_key
             .into_media()
             .ok_or(RoomSendQueueError::StorageError(RoomSendQueueStorageError::InvalidParentKey))?;
 
-        // The media we just uploaded was a thumbnail, so the thumbnail shouldn't have
+        // If the previous upload was a thumbnail, it shouldn't have
         // a thumbnail itself.
-        debug_assert!(sent_media.thumbnail.is_none());
-        if sent_media.thumbnail.is_some() {
-            warn!("unexpected thumbnail for a thumbnail!");
+        if depends_on_thumbnail {
+            debug_assert!(sent_media.thumbnail.is_none());
+            if sent_media.thumbnail.is_some() {
+                warn!("unexpected thumbnail for a thumbnail!");
```

Comment:
Oh good, there is a check for that, nice.

Reply:
Yeah, I couldn't think of a better way to do this but it is essentially the same issue in the current code already.